packages/runtime/src/types.d.ts

```diff
@@ -54,10 +54,7 @@ export interface ComponentContext {
   isRootComponent: boolean
   dataSignal: Signal<ComponentData>
   triggerEvent: (event: string, data: unknown) => void
-  apis: Record<
-    string,
-    { fetch: Function; destroy: Function; update?: Function }
-  >
+  apis: ContextApi
```

Comment:
Consider changing this to
```suggestion
  apis: Record<string, ContextApi>
```